addons/stock/product.py

```diff
@@ -365,7 +373,13 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
             res[product.id] = str(product.qty_available) +  _(" On Hand")
         return res
 
-
+    def _count_reordering_rules(self, cr, uid, ids, field_names=None, arg=None, context=None):
+        res = {}
+        product_data = self.pool['stock.warehouse.orderpoint'].read_group(cr, uid, [('product_id.product_tmpl_id', 'in', ids)], ['product_id'], ['product_id'], context=context)
+        for data in product_data:
+            product = self.pool['product.product'].browse(cr, uid, data['product_id'][0], context=context)
+            res[product.product_tmpl_id.id] = res[product.product_tmpl_id.id] + data['product_id_count'] if product.product_tmpl_id.id in res.keys() else data['product_id_count']
```

Comment:
Unnecessary code here, I think.

1/ You may use setdefault instead of checking res[key] = res[key] + var if key in res or var. I let you  play with it :) .
2/ Browsing does not seem necessary.